src/utils/report-error.ts

```diff
@@ -1,44 +1,20 @@
-import raven from 'lib/raven';
+// This function is used to report errors to Sentry.
+// This uses the `reportError` function from the `window.guardian.modules.sentry` object.
+const reportError = window.guardian.modules.sentry.reportError;
 
-type FrontendError = (Error & { reported?: boolean }) | string;
+type ErrorReportingFunction<T> = (event: T) => void;
 
-const convertError = (err: unknown): Error => {
-	if (err instanceof Error) {
-		return err;
-	}
-	if (typeof err === 'string') {
-		return new Error(err);
-	}
-	return new Error(String(err));
+const wrapWithErrorReporting = <T>(
+	fn: ErrorReportingFunction<T>,
+	tags: Record<string, string> = {},
+): ErrorReportingFunction<T> => {
+	return function (event: T) {
+		try {
+			fn(event);
+		} catch (e) {
+			reportError(e, 'on-slot-render ', { tags: JSON.stringify(tags) });
```

Comment:
I don't think we want to be passing the feature of `on-slot-render` here as this is a generic function and not related to `on-slot-render`

Reply:
Have fixed this in my commits